deployment_scripts/linode-marketplace-jupyterlab/jupyterlab-deploy.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+set -e
+trap "cleanup $? $LINENO" EXIT
+
+## JupyterLab Settings
+#<UDF name="soa_email_address" label="Email address (for the Let's Encrypt SSL certificate)" example="user@domain.tld">
+
+## Linode/SSH Security Settings
+#<UDF name="username" label="The limited sudo user to be created for the Linode" default="">
+#<UDF name="password" label="The password for the limited sudo user" example="an0th3r_s3cure_p4ssw0rd" default="">
+#<UDF name="pubkey" label="The SSH Public Key that will be used to access the Linode" default="">
+#<UDF name="disable_root" label="Disable root access over SSH?" oneOf="Yes,No" default="No">
+
+## Domain Settings
+#<UDF name="token_password" label="Your Linode API token. This is needed to create your server's DNS records" default="">
+#<UDF name="subdomain" label="Subdomain" example="The subdomain for the DNS record: www (Requires Domain)" default="">
+#<UDF name="domain" label="Domain" example="The domain for the DNS record: example.com (Requires API token)" default="">
+
+# git repo
+export GIT_REPO="https://github.com/akamai-compute-marketplace/marketplace-apps.git"
+export WORK_DIR="/tmp/marketplace-apps" 
+export MARKETPLACE_APP="apps/linode-marketplace-jupyterlab"
+
+# enable logging
+exec > >(tee /dev/ttyS0 /var/log/stackscript.log) 2>&1
+
+function cleanup {
+  if [ -d "${WORK_DIR}" ]; then
+    rm -rf ${WORK_DIR}
+  fi
+
+}
+
+function udf {
+  local group_vars="${WORK_DIR}/${MARKETPLACE_APP}/group_vars/linode/vars"
+  echo "webserver_stack: lemp" >> ${group_vars};
+  
+  if [[ -n ${USER_NAME} ]]; then
+    echo "username: ${USER_NAME}" >> ${group_vars};
+  else echo "No username entered";
+  fi
+
+  if [ "$DISABLE_ROOT" = "Yes" ]; then
+    echo "disable_root: yes" >> ${group_vars};
+  else echo "Leaving root login enabled";
+  fi
+
+  if [[ -n ${PASSWORD} ]]; then
+    echo "password: ${PASSWORD}" >> ${group_vars};
+  else echo "No password entered";
+  fi
+
+  if [[ -n ${PUBKEY} ]]; then
+    echo "pubkey: ${PUBKEY}" >> ${group_vars};
+  else echo "No pubkey entered";
+  fi
+
+  # domain vars
+  
+  if [[ -n ${SOA_EMAIL_ADDRESS} ]]; then
+    echo "soa_email_address: ${SOA_EMAIL_ADDRESS}" >> ${group_vars};
+  fi
+
+  if [[ -n ${DOMAIN} ]]; then
+    echo "domain: ${DOMAIN}" >> ${group_vars};
+  else
+    echo "default_dns: $(hostname -I | awk '{print $1}'| tr '.' '-' | awk {'print $1 ".ip.linodeusercontent.com"'})" >> ${group_vars};
+  fi
+
+  if [[ -n ${SUBDOMAIN} ]]; then
+    echo "subdomain: ${SUBDOMAIN}" >> ${group_vars};
+  else echo "subdomain: www" >> ${group_vars};
+  fi
+ 
+  if [[ -n ${TOKEN_PASSWORD} ]]; then
+    echo "token_password: ${TOKEN_PASSWORD}" >> ${group_vars};
+  else echo "No API token entered";
+  fi
+
+}
+
+function create_jupyter_token_script {
```

Comment:
capturing creds should be moved into `/roles/post/tasks/main.yml` and completed with Ansible.